server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java

```diff
@@ -450,6 +437,9 @@ synchronized void updateRemoteCluster(
             connectionMap.put(clusterAlias, remote);
             remote.ensureConnected(listener.map(ignored -> RemoteClusterConnectionStatus.RECONNECTED));
         } else {
+            if (crossProjectEnabled == false) {
```

Comment:
Is this a new change? The existing code does not seem to check this when updating it? I remember we discussed previously that `skipUnavailable` should not be used in CPS. But I forgot exactly what implementation we settled with. It feels that we should block setting it in the first place instead of silently ignore it?